notebooks/206-vision-paddlegan-anime/206-vision-paddlegan-anime.ipynb

```diff
@@ -0,0 +1,614 @@
+   "metadata": {},
+   "source": [
+    "### Convert to ONNX\n",
+    "\n",
+    "Exporting to ONNX requires specifying an input shape with PaddlePaddle's `InputSpec` and calling `paddle.onnx.export`. We check the input shape of the transformed image and use that as input shape for the ONNX model. Exporting to ONNX should not take long. If exporting succeeded, the output of the next cell will include `ONNX model saved in paddlegan_anime.onnx`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "926ca201-392b-4f19-9c9f-2de766378f35",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "target_height, target_width = transformed_image.shape[1:]\n",
+    "target_height, target_width"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "c6735bf0",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "predictor.generator.eval()\n",
+    "x_spec = InputSpec([None, 3, target_height, target_width], \"float32\", \"x\")\n",
+    "paddle.onnx.export(predictor.generator, str(model_path), input_spec=[x_spec], opset_version=11)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "93f8c13a",
+   "metadata": {},
+   "source": [
+    "### Convert to IR\n",
+    "\n",
+    "The OpenVINO IR format allows storing the preprocessing normalization in the model file. It is then no longer necessary to normalize input images manually. Let's check the transforms that the `.run()` function used:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "a35a7570-6b66-48cb-ac41-d4e7887cccd0",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "predictor.__init__??"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "51965742-8aad-4f44-beea-3f4d5a73fccb",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "t = predictor.transform.transforms[0]\n",
+    "t.params"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "3e1aef64-9bfe-4c5c-82ea-3f0399b5791a",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "## Uncomment the line below to see the documentation and code of the ResizeToScale function\n",
+    "# t??"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "03050508-a108-47fd-a95b-7a9f9e6c84a0",
+   "metadata": {},
+   "source": [
+    "There are three transforms: resize, transpose, and normalize, where normalize uses a mean and scale of [127.5,127.5,127.5]. \n",
+    "\n",
+    "The ResizeToScale function is called with (256,256) as argument for size. Further inspection of the function shows that this is\n",
+    "the minimum size to resize to. The ResizeToScale transform resizes images to the size specified in the ResizeToScale params, with width and height\n",
+    "as multiples of 32.\n",
+    "\n",
+    "Now that we know the mean and standard deviation values, and the shape of the model inputs, we can call Model Optimizer and convert the model to IR with these values. "
```

Comment:
I got lost here. I understand that the function run() calls the transforms subsequently. Is that right? What args do they need?
for resize I understand that the width and height, where the minimum value is (256,256) 
for transpose I don't see any description
for normalize we need mean and scale [127.5, 127.5, 127.5] why the scale has 3 args?